### PR TITLE
Change s:packages from list to dictionary

### PR DESCRIPTION
**Pull request by ryoppippi**
This change improves performance, especially in jetpack#tap and jetpack#get

### CODE REVIEW

**Comment by tani** [comment]
Thank you for the contribution. First, fix the test code to pass GitHub actions, please.
Next, I would like to hear the comment from @hrsh7th who is more knowledgeable developer about vim performance. At least for me, I have a possible impression to accept this idea if jetpack runs faster.  @hrsh7th Would you mind to review this idea?

**Comment by ryoppippi** [comment]
Ok I'll fix test. 
Thank you for reviewing my codes

**Comment by ryoppippi** [comment]
Ok I'll fix test codes first. 
Thanks for reviewing!

File: autoload/jetpack.vim [hunk 1]
```diff
@@ -506,23 +509,15 @@ function! jetpack#end() abort
 endfunction
 
 function! jetpack#tap(name) abort
-  for pkg in s:packages
-    if pkg.name ==# a:name
-      return isdirectory(pkg.path)
-    endif
-  endfor
-  return 0
+  return has_key(s:packages, a:name) ? isdirectory(jetpack#get(a:name).path) : 0
```
Comment (tani) [inline autoload/jetpack.vim:512]: `... ? ... : ...` is redundant.  Consider to use `&&` instead of that.

File: autoload/jetpack.vim [hunk 2]
```diff
@@ -36,7 +36,7 @@ let g:jetpack#copy_method =
   " symlink  : fs_symlink (nvim only)
   " hardlink : fs_link (nvim only)
 
-let s:packages = []
+let s:packages = {}
```
Comment (tani) [inline autoload/jetpack.vim:39]: The 'name' property is redundant. Please consider dropping it.
`for ... in values(...)` can be replaced by `for [key, value] in items(...)`.

File: autoload/jetpack.vim (hunk 1 above)
Comment (ryoppippi) [reply autoload/jetpack.vim:512]: I think it is readable, but if you think it is redundant, I'll fix it.

File: autoload/jetpack.vim (hunk 2 above)
Comment (ryoppippi) [reply autoload/jetpack.vim:39]: OK!

**Comment by ryoppippi** [comment]
I'll convert it as draft to fix test issue

**Comment by ryoppippi** [comment]
OK, I fixed the tests.
Please check it out!
@tani

**Comment by ryoppippi** [comment]
Okay.. another test error happend. Let me check.

**Comment by ryoppippi** [comment]
I fixed it.
Now it works.

**Comment by tani** [comment]
Thank you!

**Comment by ryoppippi** [comment]
Thank you too!

**Pull request merged**